Source/Core/DolphinQt/Host.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QMutex>
+#include <QObject>
+#include <QSize>
+
+// Singleton that talks to the Core via the interface defined in Core/Host.h.
+// Because Host_* calls might come from different threads than the MainWindow,
+// the Host class communicates with it via signals/slots only.
+
+// Many of the Host_* functions are ignored, and some shouldn't exist.
```

Comment:
The Core takes control of the GUI in a lot of ways, due to it directly capturing keypress/mousepress and drawing into the RenderWidget? The two are not separate things, and you can't expect them to be treated as such.